ts/sagas/profile.ts

```diff
@@ -324,17 +328,27 @@ function* handleRemoveAccount() {
 function* checkStoreHashedFiscalCode(
   profileLoadSuccessAction: ActionType<typeof profileLoadSuccess>
 ) {
-  const checkIsDifferentFiscalCode: boolean | undefined = yield select(
+  const isDifferentIdentity: CrossSessionProfileIdentity = yield select(
     isDifferentFiscalCodeSelector,
     profileLoadSuccessAction.payload.fiscal_code
   );
-  // the current logged user has a different fiscal code from the stored hashed one
-  if (checkIsDifferentFiscalCode === true) {
-    yield put(differentProfileLoggedIn());
-  }
+
   yield put(
     setProfileHashedFiscalCode(profileLoadSuccessAction.payload.fiscal_code)
   );
+
+  if (
+    [
+      CrossSessionProfileIdentity.DifferentIdentity,
+      CrossSessionProfileIdentity.Unknown
+    ].includes(isDifferentIdentity)
+  ) {
```

Comment:
@Undermaken Why don't you want `differentProfileLoggedIn()` to be dispatched in case of an `Unknown` response? All it does is reset `payments` and `messagesStatus` slices, isn't this acceptable in case of an unknown CF since the state is going to be already empty?